src/Integration.Vsix.UnitTests/CFamily/VcxRequestFactoryTests.cs

```diff
@@ -53,14 +61,247 @@ public void TryGet_NoProjectItem_Null()
+                logger: logger);
+
+            var request = testSubject.TryGet(analyzedFilePath, DummyAnalyzerOptions);
+
+            request.Should().BeNull();
+
+            logger.AssertPartialOutputStringExists(nameof(NotImplementedException));
+        }
+
+        [TestMethod]
+        public void TryGet_FailureParsing_CriticalException_ExceptionThrown()
+        {
+            const string analyzedFilePath = "c:\\test.cpp";
+
+            var fileConfig = CreateDummyFileConfig(analyzedFilePath);
+            var fileConfigProvider = SetupFileConfigProvider(DummyProjectItem, DummyAnalyzerOptions, analyzedFilePath, fileConfig.Object);
+
+            var cFamilyRulesConfigProvider = new Mock<ICFamilyRulesConfigProvider>();
+            cFamilyRulesConfigProvider
+                .Setup(x => x.GetRulesConfiguration(SonarLanguageKeys.CPlusPlus))
+                .Throws<StackOverflowException>();
+
+            var testSubject = CreateTestSubject(DummyProjectItem,
+                fileConfigProvider: fileConfigProvider.Object,
+                cFamilyRulesConfigProvider: cFamilyRulesConfigProvider.Object);
+
+            Action act = () => testSubject.TryGet(analyzedFilePath, DummyAnalyzerOptions);
+
+            act.Should().ThrowExactly<StackOverflowException>();
+        }
+
+        [TestMethod]
+        public void TryGet_IRequestPropertiesAreSet()
+        {
+            var analyzerOptions = new CFamilyAnalyzerOptions();
+            var rulesConfig = Mock.Of<ICFamilyRulesConfig>();
+
+            var request = (IRequest)GetSuccessfulRequest(analyzerOptions, "d:\\xxx\\fileToAnalyze.cpp", rulesConfig);
+            request.Should().NotBeNull();
+
+            request.Context.File.Should().Be("d:\\xxx\\fileToAnalyze.cpp");
+            request.Context.PchFile.Should().Be(SubProcessFilePaths.PchFilePath);
+            request.Context.AnalyzerOptions.Should().BeSameAs(analyzerOptions);
+            request.Context.RulesConfiguration.Should().BeSameAs(rulesConfig);
+        }
+
+        [TestMethod]
+        public void TryGet_FileConfigIsSet()
+        {
+            var request = GetSuccessfulRequest();
+            request.Should().NotBeNull();
+
+            request.FileConfig.Should().NotBeNull();
+        }
+
+        [TestMethod]
+        public void TryGet_NonHeaderFile_IsSupported()
+        {
+            var request = GetSuccessfulRequest();
+            
+            request.Should().NotBeNull();
+            (request.Flags & Request.MainFileIsHeader).Should().Be(0);
+        }
+
+        [TestMethod]
+        public void TryGet_HeaderFile_IsSupported()
+        {
+            var projectItemConfig = new ProjectItemConfig { ItemType = "ClInclude" };
+            var projectItemMock = CreateMockProjectItem("c:\\foo\\xxx.vcxproj", projectItemConfig);
+
+            var fileConfig = CreateDummyFileConfig("c:\\dummy\\file.h");
+            fileConfig.Setup(x => x.CompileAs).Returns("CompileAsCpp");
```

Comment:
This was hidden in the original tests -- if you don't specify CompileAsCpp (or as C), we won't detect a language and would not analyze the header file. I guess it's ok because you need to do it for header files anyway. At least now it's explicit in the test.